### ISSUE TITLE
Create testing env without optional dependencies

### ISSUE THREAD

**Issue by cbrnr** [scot-dev/scot#166]
Our only required dependencies are numpy and scipy. Therefore, we should create a testing env in Travis which includes only these two packages, and not matplotlib, scikit-learn, and mne. This will likely result in some of our tests failing - maybe we should decorate specific tests if they require optional packages (e.g. `@require_sklearn`, `@require_matplotlib`, `@require_mne`).

**Comment by mbillingr** [comment]
I have been wondering how to skip certain tests depending on the environment. Decorators seem like a good idea.

**Comment by cbrnr** [comment]
Yeah, that's how e.g. the MNE folks do it. See https://github.com/mne-tools/mne-python/blob/master/mne/utils.py around line 700.
